### PR TITLE
bpf: Retire CT entries quickly for unreplied connections 

### PR DESCRIPTION
**Pull request by joestringer**
Set a five minute timeout for conntrack entries that have not yet seen a
non-SYN packet. For non-TCP protocols, the timeouts remain longer. ICMP
response tuples for TCP connections will be timed out quickly in all cases.

Enable active garbage collection of CT LRU maps from userspace. 

Fixes: #3232
Related: #3236

### CODE REVIEW

**Comment by joestringer** [comment]
test-me-please

File: bpf/lib/conntrack.h [hunk 1]
```diff
@@ -51,9 +52,13 @@ static inline void __inline__ __ct_update_timeout(struct ct_entry *entry,
...
 
-static inline void __inline__ ct_update_timeout(struct ct_entry *entry)
+static inline void __inline__ ct_update_timeout(struct ct_entry *entry, bool syn)
 {
-	__ct_update_timeout(entry, CT_DEFAULT_LIFETIME);
+	entry->seen_non_syn |= !syn;
+	if (entry->seen_non_syn)
+		__ct_update_timeout(entry, CT_DEFAULT_LIFETIME);
+	else
+		__ct_update_timeout(entry, CT_SYN_TIMEOUT);
```
Comment (tgraf) [inline bpf/lib/conntrack.h:61]: `__ct_update_timeout()` is protected by `NEEDS_TIMEOUT` which is undefined if we are running with LRU. The expiration needs to run with LRU as well.

Comment (joestringer) [reply bpf/lib/conntrack.h:61]: Good point, I added another commit to enable these timeouts in LRU case and to enable garbage collection in userspace for conntrack maps backed by LRU map type.

**Comment by joestringer** [comment]
test-me-please

**Comment by joestringer** [comment]
test-me-please

**Comment by tgraf** [comment]
test-me-please

**Comment by tgraf** [comment]
test-me-please

**Comment by tgraf** [comment]
test-me-please

**Comment by joestringer** [comment]
test-me-please

**Comment by joestringer** [comment]
test-me-please

**Comment by joestringer** [comment]
test-me-please

**Comment by joestringer** [comment]
test-me-please

**Pull request merged**